### PR TITLE
User name [Don't Merge]

### LINKED ISSUE #73: Navigation bar
**Issue Deadline:**
- 06/17/2024

**Issue Description:**
- User name displaying when the user is logged in and implementing the styles to it based on the designs

**Acceptance Criteria:**
- Once the user logs in, the navigation bar should display their name
- The drop-down where the user name is displayed should have logout and profile page links

**Time Estimation:**
- Approximately 10

**Sub-tasks:**
- Creating a route to get the user's profile (username) information

### PR DESCRIPTION
**Pull request by Nemo080**
**Backend/Server**
- users route created
- authMiddleware code edited to use jwt for authentication
- auth route for signup and login fixed so that the authentication takes a jwt token and not a string

**Frontend**
- User name displays in the navbar
- layout is used to change between navbars when the user logs in
- Applayout handles the user's information that's retrieved from the user route

### CODE REVIEW

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -34,15 +39,23 @@ function App() {
   const handleLogout = () => {
     setIsLoggedIn(false);
     localStorage.setItem('isLoggedIn', 'false');
+    window.location.href = '/login';
```
Comment (joselevelsup) [inline frontend/src/App.jsx]: Use React Router's Hook, `useNavigate` to handle navigation programmatically.

File: frontend/src/components/Signup.jsx [hunk 2]
```diff
@@ -1,82 +1,93 @@
...
+  const [email, setEmail] = useState();
+  const [password, setPassword] = useState();
+  const navigate = useNavigate();
 
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    axios
+      .post("http://localhost:3000/auth/signup", { name, email, password })
+      .then((result) => {
+        console.log(result);
```
Comment (joselevelsup) [inline frontend/src/components/Signup.jsx]: Remove Console.logs

File: frontend/src/components/Signup.jsx [hunk 3]
```diff
@@ -1,82 +1,93 @@
...
+        const userToken = "userToken";
+        localStorage.setItem(userToken, result.data.token);
+        const token = localStorage.getItem(userToken);
+        if (token) {
+          console.log("token collected");
```
Comment (joselevelsup) [inline frontend/src/components/Signup.jsx]: remove

File: frontend/src/components/Signup.jsx [hunk 4]
```diff
@@ -1,82 +1,93 @@
...
+          handleLogin();
+          navigate("/app/user-profile");
+        } else {
+          console.log("token not collected");
```
Comment (joselevelsup) [inline frontend/src/components/Signup.jsx]: remove

File: server/routes/users.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import express from 'express';
+import prisma from '../db/index.js';
+
+export default function (passport){
+    const router = express.Router();
+
+
+router.get("/me", passport.authenticate("jwt", {session:false}), async function (req, res){
+    console.log(req.user)
```
Comment (joselevelsup) [inline server/routes/users.js]: remove

File: frontend/src/App.jsx (hunk 1 above)
Comment (Nemo080) [reply frontend/src/App.jsx]: Has been resolved

File: frontend/src/components/Signup.jsx (hunk 2 above)
Comment (Nemo080) [reply frontend/src/components/Signup.jsx]: Has been removed

File: frontend/src/components/Signup.jsx (hunk 3 above)
Comment (Nemo080) [reply frontend/src/components/Signup.jsx]: Has been removed

File: frontend/src/components/Signup.jsx (hunk 4 above)
Comment (Nemo080) [reply frontend/src/components/Signup.jsx]: Has been removed

File: server/routes/users.js (hunk 5 above)
Comment (Nemo080) [reply server/routes/users.js]: Has been removed

**Pull request merged**